tools/checkcomments/checkcomments.go

```diff
@@ -56,36 +56,35 @@ func main() {
 func run(pass *analysis.Pass) (any, error) {
 	var iCache issueCache
 
-	current_path, err := os.Getwd()
+	currentPath, err := os.Getwd()
 	if err != nil {
-		log.Fatal(err)
+		return nil, err
 	}
 
-	cache_path := getCacheFilePath(current_path)
+	cachePath := getCacheFilePath(currentPath)
 
-	cf, err := lockedfile.OpenFile(cache_path, os.O_RDWR|os.O_CREATE, 0o666)
+	cf, err := lockedfile.OpenFile(cachePath, os.O_RDWR|os.O_CREATE, 0o666)
 	if err != nil {
-		log.Fatal(err)
+		return nil, err
 	}
 
 	defer func() {
-		cerr := cf.Close()
-		if cerr != nil {
-			log.Fatal(cerr)
+		if err = cf.Close(); err != nil {
+			log.Println(err)
 		}
 	}()
```

Comment:
Regarding the first one, if we exit from `run` with some error, e.g. smth related to github, the initial error is more important than an error that could be returned if the file wasn't closed. 

Regarding the second one, yes, I replaced all `log.Fatal`s because of this. And also `log.Fatal` exists immediately which is not super nice when there is something to `defer` :)

Reply:
> could make the code a bit cleaner

But error messages less helpful.